tests/Http/PostTest.php

```diff
@@ -718,17 +718,24 @@ public function testPostsIndexAsOwnerHiddenPosts()
 
         // Create a hidden post from the same $ownerId.
         $hiddenPost = factory(Post::class, 'accepted')->create();
-        $hiddenPost->tag('Hide In Gallery');
         $hiddenPost->northstar_id = $ownerId;
         $hiddenPost->save();
 
+        $this->withAdminAccessToken()->postJson('api/v3/posts/' . $hiddenPost->id . '/tags', [
+            'tag_name' => 'Hide In Gallery',
+        ]);
+
         // Create anothter hidden post by different user.
         $secondHiddenPost = factory(Post::class, 'accepted')->create();
-        $secondHiddenPost->tag('Hide In Gallery');
         $secondHiddenPost->northstar_id = $this->faker->unique()->northstar_id;
         $secondHiddenPost->save();
 
+        $this->withAdminAccessToken()->postJson('api/v3/posts/' . $secondHiddenPost->id . '/tags', [
+            'tag_name' => 'Hide In Gallery',
+        ]);
+
         $response = $this->withAccessToken($ownerId)->getJson('api/v3/posts');
```

Comment:
@DFurnes this seems to fail bc I _think_ it is still using the `AdminAccessToken` and is actually grabbing hidden posts we don't want to see. Is there a way to make sure it's using the `withAccessToken($ownerId)` instead of the `withAdminAccessToken()` first used?

Reply:
A ha!! I think you're totally right that we'd be keeping the same cached user in the guard! 

We should maybe add an `Auth::logout()` to [the Gateway guard](https://github.com/DoSomething/gateway/blob/master/src/Server/GatewayGuard.php) so that we can unset that property in tests (since `Auth::setUser` is type-hinted to require an actual user instance).

Reply:
awesome @DFurnes ! Here's a [the PR](https://github.com/DoSomething/gateway/compare/logout?expand=1) !